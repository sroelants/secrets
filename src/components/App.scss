@import './common';

html {
    overflow: scroll;
    overflow-x: hidden;
}

@media (max-width: $bp-small) {
    html {
        font-size: 0.7rem;
    }
}

body {
    background: $white;
    color: $black;
    font-family: $sansfont;
}

// Main [MAIN] -----------------------------------------------------------------

.footer {
    display: flex;
    align-items: center;
    justify-content: center;
    min-height: 150px;

    a {
        font-weight: 700;
        text-decoration: none;
        color: inherit;
        font-style: italic;
    }
}
