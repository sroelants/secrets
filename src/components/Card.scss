@import "./common";

.card {
  font-size: 1.5rem;
  min-height: 5rem;
  padding: 1rem;
  border: 1px solid $lgrey;
  border-radius: 2px;
  background: $white;
  box-shadow: 6px 6px 15px rgba(0, 0, 0, 0.03), 2px 4px 4px rgba(0, 0, 0, 0.05),
    0 0 10px rgba(0, 0, 0, 0.05);
  animation: slide-in;
  animation-duration: 0.3s;
  width: 90%;
  margin: 30px auto;
}

@media (min-width: $bp-medium) {
  .card {
    font-size: 1.3rem;
    width: 50%;
  }
}

@keyframes slide-in {
  from {
    opacity: 0;
    transform: translateY(100px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

.card__header {
  font-size: 0.8em;
  color: $grey;
  display: flex;
  justify-content: space-between;
}

.card__date {
  text-transform: uppercase;
}

.card__secret {
  margin-top: 10px;
  font-family: $seriffont;
  font-size: 1em;
  line-height: 1.5;
}

@media (min-width: $bp-small) {
  .card__secret {
    font-size: 0.9em;
  }
}

.card__like {
  width: 1em;
  filter: grayscale(1);
  opacity: 0.7;
  transition: all 0.3s linear;
  cursor: pointer;
  margin: 0 10px 0 10px;
  position: relative;
  top: 3px;
}

.liked {
  opacity: 1;
  filter: grayscale(0);
  animation: bounce;
  animation-duration: 0.2s;
}

@keyframes bounce {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(0.5);
  }
  100% {
    transform: scale(1);
  }
}
