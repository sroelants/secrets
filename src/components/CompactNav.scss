@import "common";

.overlay {
  box-sizing: border-box;
  position: fixed;
  height: 100%;
  width: 100%;
  background-color: $white;
  opacity: 0;
  top: 0;
  left: 0;
  z-index: 5;
  overflow: none;
  // transform: translateX(100%);
  visibility: hidden;
  transition: all 0.3s linear;
  &.overlay--visible {
    visibility: visible;
    opacity: 0.9;
  }
}

.nav__hamburger {
  cursor: pointer;
  display: flex;
  flex-direction: column;
  height: 30px;
  justify-content: space-between;
  position: absolute;
  right: 50px;
  top: 30px;
  width: 40px;
  z-index: 20;
  &:hover > .nav__hamburger__bar {
    background-color: $grey;
  }

  &:active > .nav__hamburger__bar {
    background-color: $lgrey;
  }
}

.nav__hamburger__bar {
  background-color: $black;
  height: 4px;
  border-radius: 2px;
  transition: all 300ms cubic-bezier(0.785, 0.135, 0.15, 0.86); /* easeInOutCirc */
  opacity: 1;
  transform: none;
}

.sidemenu {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: right;
  background-color: $white;
  height: 100vh;
  padding-left: 50px;
  padding-right: 25px;
  position: fixed;
  top: 0;
  right: 0;
  visibility: hidden;
  transform: translateX(100%);
  z-index: 10;

  .nav__links {
    display: flex;
    flex-direction: column;
    &:hover:before {
      visibility: hidden;
    }
  }

  .nav__link {
    font-size: 2.5rem;
    text-align: right;
  }
  &.sidemenu--visible {
    animation: 0.2s ease-in-out slide-right;
    visibility: visible;
    transform: translateX(0);
    box-shadow: 6px 6px 15px rgba(0, 0, 0, 0.03),
      4px 2px 4px rgba(0, 0, 0, 0.05), 0 0 10px rgba(0, 0, 0, 0.05);
  }
}

@keyframes slide-right {
  from {
    transform: translateX(100%);
  }
  to {
    transform: translateX(0);
  }
}

// Hamburger animation
.nav__hamburger__top-bar {
  transform-origin: left top;
}

.nav__hamburger__bottom-bar {
  transform-origin: left bottom;
}

.nav__hamburger--open > .nav__hamburger__top-bar {
  transform: rotate(45deg);
}
.nav__hamburger--open > .nav__hamburger__bottom-bar {
  transform: rotate(-45deg);
}

.nav__hamburger--open > .nav__hamburger__middle-bar {
  opacity: 0;
}
