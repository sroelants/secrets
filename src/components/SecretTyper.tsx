import React, { useEffect, useState } from 'react';

const secrets: string[] = [
  `A month after being raped, I went and seduced my rapist.
   I thought if it was my choice and I initiated the sex it
   would make things better. It didnâ€™t.`,
  `My youth group is one of the things keeping me from killing myself.
  I'm the youth pastor.`,
  `Last week when I was shopping, the lady ahead of me dropped a
   $20 bill on the ground.  Instead of saying anything, I quickly
   stepped on it and claimed it as my own.`];

export const SecretTyper: React.FC = () => {
  const [text, setText] = useState('');
  const [isDeleting, setIsDeleting] = useState(false);
  const [delay, setDelay] = useState(50);
  const [loop, setLoop] = useState(0);

  const i: number = loop % secrets.length
  const secret = secrets[i];

  const handleTyping = () => {
    // Change text and speed.
    setText(
      isDeleting
        ? secret.substring(0, text.length - 1)
        : secret.substring(0, text.length + 1)
    );

    // Get new delay
    setDelay(
      isDeleting
        ? 20
        : text[text.length - 1] === '.'
          ? 400 // Pause at a period for realism.
          : 40
    );

    // Test whether finished typing
    if (!isDeleting && text === secret) {
      // Pause and start deleting
      setTimeout(() => setIsDeleting(true), 1000);
    }

    // Test whether finished deleting
    if (isDeleting && text === "") {
      // Done deleting, load new secret (by increasing loop number)
      setIsDeleting(false);
      setLoop(loop + 1);
    }
  };

  useEffect(() => {
    const timer = setTimeout(handleTyping, delay);
    return () => clearInterval(timer)
  });
  return <div className="header__secret"> {text} <span id="cursor" /></div>
}
