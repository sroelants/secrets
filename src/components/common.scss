$white: #fafafa;
$black: #444;
$lblack: #555;
$grey: #888;
$lgrey: hsla(0, 0, 60, 0.3);
$sansfont: "Nunito", sans-serif;
$seriffont: "Libre Baskerville", serif;

$bp-small: 580px;
$bp-medium: 900px;
$bp-wide: 1200px;
