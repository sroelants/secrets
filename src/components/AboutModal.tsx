import './Modal.scss';
import React from 'react';
import { ModalType } from './App'

export interface IModal {
  visible: boolean;
  closeHandler: (t: ModalType | null) => void;
}

export const AboutModal: React.FC<IModal> = ({ visible, closeHandler }) => {
  return (
    <>
      <div className={visible ? "about-modal modal visible" : "about-modal modal"} >
        <button className="modal__close" onClick={() => {
          closeHandler(null);
        }}>
        <img aria-label="close"
          src="assets/img/close.svg"
          width="25px"
          alt="close" />
        </button>
        <div className="modal__title">About Secrets</div>
        <div className="modal__content">
          <p>
          We all know that feeling of relief after sharing a long-kept secret
          with someone. A weight is lifted. What seemed like a terrible or
          embarrasing thing all of a sudden doesn't seem that important anymore.
          Sharing secrets with people brings us perspective. We're not only
          opening ourselves up to them, we're opening up to ourselves.
        </p>

        <p>
          The science <a href="https://psycnet.apa.org/record/2017-20428-001"
          target="_blank" rel="noopener noreferrer"> has
          long been in </a> on the benefits of sharing secrets with
          others. Keeping secrets is a mental, emotional and physical burden.
          Our minds keep drifting and focussing on them. We are more anxious
          and adrenaline levels and blood pressure are above average.
        </p>
        <p>
          On the flip side, sharing secrets with others has a tremendous effect
          on us. Even if we're uncomfortable sharing with someone close to us,
          for fear of judgement, we ofter share our deepest secrets with people
          we're not close to, even complete strangers like a hairdresser or
          taxi driver. Sharing secrets forms bonds of trust, they strengthen
          social cohesion, even between strangers. Aside from benefiting us
          emotionally and physically, sharing secrets benefits our society as a
          hole.
        </p>
        <p>
          This page is meant as a platform for people to share their secrets
          with strangers, completely anonymously. Perhaps the person sitting
          next to you on the bus tomorrow will know and sympathize with
          something you haven't yet had the courage to face up to, and that is
          an encouraging thought.
        </p>
        </div>
      </div >
      <div className="dimmer"></div>
    </>
  );
};
