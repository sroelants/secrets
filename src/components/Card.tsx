import React, {useState } from "react";
import "./Card.scss";

export interface ICard {
  secretId: number;
  secret: string;
  date: string;
  likes: number;
}

export const Card: React.FC<ICard> = ({ secretId, secret, date, likes }) => {
  // Check if secret is already liked (in localStorage)
  const storedLike = localStorage.getItem(secretId.toString());
  let alreadyLiked: boolean;
  if (storedLike === null) {
    alreadyLiked = false;
  } else {
    alreadyLiked = true;
  }

  const [liked, setLiked] = useState(alreadyLiked);
  const [localLikes, setLocalLikes] = useState(likes);
  const dateObject = new Date(date);
  const shortDate = dateObject
    .toString()
    .split(" ")
    .slice(1, 4)
    .join(" ");

  const handleLike = () => {
    // Send put request to server.
    const update = {
      date_posted: date,
      likes: !liked ? likes + 1 : likes - 1,
      secret,
    };

    fetch("https://secrets-backend.herokuapp.com/api/secret/" + secretId, {
      body: JSON.stringify(update),
      headers: new Headers({ "Content-Type": "application/json" }),
      method: "PUT",
    });

    // Set current secret's like status in localStorage
    if (liked) {
      localStorage.removeItem(secretId.toString());
    } else {
      localStorage.setItem(secretId.toString(), "liked");
    }

    setLocalLikes(localLikes + (!liked ? 1 : -1)); // Update shown like count.
    setLiked(!liked); // Update shown like status
  };

  return (
    <div className="card">
      <div className="card__header">
        <div className="card__date">{shortDate}</div>
        <div className="card__likes">
          {localLikes}
          <img alt="like"
            className={liked ? "card__like liked" : "card__like"}
            src="/assets/img/like.svg"
            onClick={handleLike}
          />
        </div>
      </div>
      <div className="card__secret">{secret}</div>
    </div>
  );
};
