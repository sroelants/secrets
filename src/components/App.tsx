import "./App.scss";
import React, { useState } from "react";
import { AboutModal } from "./AboutModal";
import { Header, IHeader } from "./Header";
import { Secrets } from "./Secrets";
import { ShareModal } from "./ShareModal";

export enum ModalType {
  About,
  Share,
}

const App: React.FC = () => {
  const [modalState, setModalState] = useState<ModalType | null>(null);

  const setModalHandler = (mode: ModalType | null) => {
    setModalState(mode);
  };

  const HeaderProps: IHeader = { clickHandler: setModalHandler };

  return (
    <div className="App">
      <Header {...HeaderProps} />
      <main className="main">
        <Secrets />
      </main>
      <AboutModal visible={modalState === ModalType.About}
        closeHandler={setModalHandler} />
      <ShareModal visible={modalState === ModalType.Share}
        closeHandler={setModalHandler} />
      <footer className="footer">
        <p>
        Created by
        <a href="https://www.samroelants.com"
          target="_blank"
          rel="noopener noreferrer">
           &nbsp;Sam Roelants
        </a>
      </p>
      </footer>
    </div>
  );
};

export default App;
