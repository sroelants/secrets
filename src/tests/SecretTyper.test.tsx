import React from "react";
import renderer from "react-test-renderer";
import {SecretTyper} from "../components/SecretTyper";
import {shallow, mount} from "enzyme";

// Unit
describe("SecretTyper", () => {
  it("renders without crashing", () => {
    shallow(<SecretTyper />);
  });

  it("Matches the snapshot", () => {
    const output = renderer.create(<SecretTyper />);
    expect(output).toMatchSnapshot();
  });
});
