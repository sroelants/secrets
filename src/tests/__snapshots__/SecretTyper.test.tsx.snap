// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecretTyper Matches the snapshot 1`] = `
<div
  className="header__secret"
>
   
  
   
  <span
    id="cursor"
  />
</div>
`;
